internal/ingress/controller/store/store_test.go

```diff
@@ -740,3 +740,76 @@ func TestUpdateSecretIngressMap(t *testing.T) {
 		}
 	})
 }
+
+// Test to avoid regression fixed in https://github.com/kubernetes/ingress-nginx/pull/2361
+func TestListIngressWithoutPath(t *testing.T) {
+	s := newStore(t)
+
+	ing1 := &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "testns",
+		},
+		Spec: v1beta1.IngressSpec{
+			Backend: &v1beta1.IngressBackend{
+				ServiceName: "demo",
+				ServicePort: intstr.FromInt(80),
+			},
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "foo.bar",
+				},
+			},
+		},
+	}
+	s.listers.Ingress.Add(ing1)
+
+	ing2 := &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test2",
+			Namespace: "testns",
+			Annotations: map[string]string{
+				"kubernetes.io/ingress.class": "something",
+			},
+		},
+		Spec: v1beta1.IngressSpec{
+			Backend: &v1beta1.IngressBackend{
+				ServiceName: "demo",
+				ServicePort: intstr.FromInt(80),
+			},
+		},
+	}
+	s.listers.Ingress.Add(ing2)
+
+	ing3 := &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test3",
+			Namespace: "testns",
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "foo.bar",
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								{
+									Backend: v1beta1.IngressBackend{
+										ServiceName: "demo",
+										ServicePort: intstr.FromInt(80),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	s.listers.Ingress.Add(ing3)
+
```

Comment:
For completeness I'd also add another ingress similar to `ing2` but this time with the default supported class of `nginx`.

Reply:
done